core/src/shred_fetch_stage.rs

```diff
@@ -14,9 +16,54 @@ pub struct ShredFetchStage {
 }
 
 impl ShredFetchStage {
+    // updates packets received on a channel and sends them on another channel
+    fn modify_packets<F>(recvr: PacketReceiver, sendr: PacketSender, modify: F)
+    where
+        F: Fn(&mut Packet),
+    {
+        while let Some(mut p) = recvr.iter().next() {
+            p.packets.iter_mut().for_each(|p| modify(p));
+            if sendr.send(p).is_err() {
+                break;
+            }
+        }
+    }
+
+    fn packet_modifier<F>(
+        sockets: Vec<Arc<UdpSocket>>,
+        exit: &Arc<AtomicBool>,
+        sender: PacketSender,
+        recycler: Recycler<PinnedVec<Packet>>,
+        modify: F,
+    ) -> (Vec<JoinHandle<()>>, JoinHandle<()>)
+    where
+        F: Fn(&mut Packet) + Send + 'static,
+    {
+        let (packet_sender, packet_receiver) = channel();
+        let streamers = sockets
+            .into_iter()
+            .map(|s| {
+                streamer::receiver(
```

Comment:
Why not modify the `streamer::receiver` to directly take the "modify" function instead of adding an intermediate modifier?

Reply:
This modifying business really doesn't belong in the streamer.